tfrecorder/schema.py

```diff
@@ -88,52 +96,48 @@ def get_raw_schema_map(
   """Converts a schema to a raw (pre TFT / post image extraction) schema."""
   raw_schema = {}
   for k, v in schema_map.items():
-    if v.type_name == "image_uri":
-      raw_schema['image_name'] = image_support_type
-      raw_schema['image'] = image_support_type
-      raw_schema['image_height'] = image_support_type
-      raw_schema['image_width'] = image_support_type
-      raw_schema['image_channels'] = image_support_type
+    if v.type_name == 'image_uri':
+      raw_schema['image_name'] = image_support_string
+      raw_schema['image'] = image_support_string
+      raw_schema['image_height'] = image_support_int
+      raw_schema['image_width'] = image_support_int
+      raw_schema['image_channels'] = image_support_int
     else:
       raw_schema[k] = schema_map[k]
   return raw_schema
 
 
-def get_tft_coder(columns: List[str],
-                  schema_map: Dict[str, collections.namedtuple]
-                  ) -> tft.coders.CsvCoder:
+def get_tft_coder(
+    columns: List[str],
+    schema_map: Dict[str, collections.namedtuple]
+    ) -> tft.coders.CsvCoder:
   """Gets a TFT CSV Coder.
 
   Args:
-    columns: Ordered dataframe column names, from df.column.
+    columns: Ordered DataFrame column names, from df.column.
     schema_map: Schema map used to infer the schema.
 
   Returns:
     tft.coders.CsvCoder
   """
-  feature_spec = dict()
-
+  feature_spec = {}
   # Because the DF column name order may not match the feature_spec order
-  # This maps existing column names to their feature spec (req part of
+  # This maps existing column names to their feature spec (required part of
   # namedtuple)
   for col in columns:
     feature_spec[col] = schema_map[col].feature_spec
 
-
   metadata = dataset_metadata.DatasetMetadata(
       schema_utils.schema_from_feature_spec(feature_spec))
-  return tft.coders.CsvCoder(columns,
-                             metadata.schema)
 
+  return tft.coders.CsvCoder(columns, metadata.schema)
 
-def get_key(
-    type_name: str,
-    schema_map: Dict[str, collections.namedtuple]
+
+def get_key(type_name: str, schema_map: Dict[str, collections.namedtuple]
     ) -> Union[str, None]:
-  """Gets key of type 'type_name' from schema map if present,
-  otherwise returns None
+  """Gets key of type 'type_name' from schema map.
 
-  Used by the beam pipeline to identify if images or split are present.
+  Returns type_name if present, otherwise returns None.
```

Comment:
Shouldn't this say returns key? Also it looks like it will pick up the first key with matching `type_name`, unless all key, type_name pairs are unique.

Reply:
Yes to your first point. Very good point on the second issue, and that seems limiting.

Reply:
Added a todo and documentation for second point, based on our conversation.